sui/token_bridge/sources/create_wrapped.move

```diff
@@ -118,4 +121,38 @@ module token_bridge::create_wrapped {
             decimals,
         );
     }
+
+    public entry fun update_registered_wrapped_coin_metadata<CoinType>(
+        token_bridge_state: &mut State,
+        worm_state: &mut WormholeState,
+        vaa_bytes: vector<u8>,
+        metadata: &mut CoinMetadata<CoinType>,
+        ctx: &mut TxContext
+    ) {
+        let vaa = vaa::parse_verify_and_replay_protect(
+            token_bridge_state,
+            worm_state,
+            vaa_bytes,
+            ctx
+        );
+        let payload = core_vaa::destroy(vaa);
+        let meta = asset_meta::deserialize(payload);
+        let origin_chain = asset_meta::token_chain(&meta);
+
+        assert!(
+            origin_chain != wormhole_state::chain_id(),
+            E_UPDATING_NATIVE_COIN_META
+        );
+        assert!(
+            state::is_registered_asset<CoinType>(token_bridge_state),
+            E_WRAPPED_COIN_NOT_INITIALIZED
```

Comment:
is this check necessary? the existence of `metadata` implies it's been set up, doesn't it?

Reply:
just because the metadata exists doesn't mean it is registered with the bridge as a wrapped asset native better error message would be `E_COIN_NOT_REGISTERED_AS_WRAPPED`